flowmachine/flowmachine/features/utilities/feature_collection.py

```diff
@@ -82,53 +82,50 @@ class FeatureCollection(Query):
 
     """
 
-    def __init__(self, metrics, dropna=True):
-
-        self.joined_metrics = self.__join_queries(metrics, dropna)
-
-        super().__init__()
-
-    @classmethod
-    def from_list_of_classes(cls, classes, *args, **kwargs):
-        """
-        From uninstantiated classes with common arguments.
-        """
-
-        metrics = [c(*args, **kwargs) for c in classes]
-        return cls(metrics)
-
-    # Private method that joins multiple queries together
-    # and returns a joined query.
-    @staticmethod
-    def __join_queries(queries, dropna):
-
-        # We want to handle the first case as a special case, as we
-        # need to give the left object a name on the first join, but
-        # not in any subsequent joins.
-        col = queries[0].column_names[0]
-        left_append = "_" + queries[0].__class__.__name__ + "_0"
-        right_append = "_" + queries[1].__class__.__name__ + "_1"
-        how = "inner" if dropna else "full outer"
-        running_join = queries[0].join(
-            queries[1],
-            on_left=col,
-            left_append=left_append,
-            right_append=right_append,
-            how=how,
-        )
-
-        for i, q in enumerate(queries[2:]):
-            col = q.column_names[0]
-            append = "_" + q.__class__.__name__ + "_{}".format(i + 2)
-            running_join = running_join.join(
-                q, on_left=col, right_append=append, how=how
-            )
-            # Trigger memoization
-            _ = q.md5
-            _ = running_join.md5
-            _ = running_join.column_names
-
-        return running_join
-
-    def _make_query(self):
-        return self.joined_metrics.get_query()
+    return __join_queries(metrics, dropna)
+
+
+def feature_collection_from_list_of_classes(
+    classes, *args, dropna=False, **kwargs
+) -> Join:
+    """
+    Create a feature collection from uninstantiated classes with common arguments.
+
+    See Also
+    --------
+    feature_collection
+    """
+
+    metrics = [c(*args, **kwargs) for c in classes]
+    return feature_collection(metrics, dropna=dropna)
+
+
+# Private method that joins multiple queries together
```

Comment:
s/method/function/`?

Also, not sure if name mangling is any issue here but maybe replace the leading double-underscore with a single one?